public/app/features/alerting/unified/components/contact-points/ContactPointHeader.tsx

```diff
@@ -26,21 +34,54 @@ interface ContactPointHeaderProps {
 export const ContactPointHeader = ({ contactPoint, disabled = false, onDelete }: ContactPointHeaderProps) => {
   const { name, id, provisioned, policies = [] } = contactPoint;
   const styles = useStyles2(getStyles);
+  const [showPermissionsDrawer, setShowPermissionsDrawer] = useState(false);
+  const { selectedAlertmanager } = useAlertmanager();
 
   const [exportSupported, exportAllowed] = useAlertmanagerAbility(AlertmanagerAction.ExportContactPoint);
   const [editSupported, editAllowed] = useAlertmanagerAbility(AlertmanagerAction.UpdateContactPoint);
   const [deleteSupported, deleteAllowed] = useAlertmanagerAbility(AlertmanagerAction.UpdateContactPoint);
-
   const [ExportDrawer, openExportDrawer] = useExportContactPoint();
 
-  const numberOfPolicies = policies.length;
-  const isReferencedByAnyPolicy = numberOfPolicies > 0;
-  const isReferencedByRegularPolicies = policies.some((ref) => ref.route.type !== 'auto-generated');
+  const showManagePermissions = showManageContactPointPermissions(selectedAlertmanager!, contactPoint);
+
+  const regularPolicyReferences = policies.filter((ref) => ref.route.type !== 'auto-generated');
```

Comment:
We should not filter out autogenerated policies for calculating the count. Take in account that only admin roles will get the autogenerated policies from the response. And in that case, I believe, admin users should count this policy in the count, as when filtering.
If we use this filtered list of polices, then there is a mismatch between what we see in the contact point list, and what we see in the policy tree after clicking to the header:

<img width="189" alt="Screenshot 2024-09-19 at 16 55 49" src="https://github.com/user-attachments/assets/52c3ccc3-813b-4b86-867a-fbf843d95827">

<img width="932" alt="Screenshot 2024-09-19 at 16 55 40" src="https://github.com/user-attachments/assets/e82a628c-a9fd-46c9-bfde-cc331ca0bb98">

Reply:
I do not think that auto-generated policies should be counted in metadata because :
1. the actual usage is covered by UsedByRules
2. neither the user nor the admin have control over those policies. They are displayed for debugging purposes. 

However, I think we should not highlight them in the search results.

Reply:
> I do not think that auto-generated policies should be counted in metadata

I agree 💯

But If we don't highlight auto-generated policies in the filter, admin users are not able to see how many alert instances (the actual total)  are being routed to a particular contact point.
Maybe we could change the text in the contact point header to something like:  `Used by one 'regular' notification policy` ?

Reply:
If this is causing confusion for users, I would lean toward not showing the autogenerated policies in the UI at all, even to admins. IMO, they are an implementation detail that we decided might be useful for admins to see. 

The fact that we create root policies for contact points that aren't used by any rules is double-so an implementation detail that exists only to prevent accidental routing notifications to the default route.

Reply:
Maybe we should put it behind a feature toggle, if we/users needed it for debugging purposes? Or maybe instead a feature toggle we could add a query parameter that would reveal the autogenerated branch in the UI?

Reply:
Based on the discussion the other day in the experience weekly, how do we want to proceed here?

Right now, I've made it so, with the feature toggle off, it behaves as before. 

With it turned on, the number of policies shown only counts those that directly mention the contact point (i.e. excluding the auto generated ones for simplified routing).

It will also show the number of rules that explicitly mention the rule via simplified routing.

Are we okay to proceed with this and to tackle the autogenerated policy display separately? WDYT?

Reply:
maybe in case of `admin` user, we can render 'used by x notification polices (autogenerated are not included) or something like this?

Reply:
Let's keep it as it is and address our concerns in a follow-up PRs